### ISSUE TITLE
feature: use StreamText.net input from a human captioner?

### ISSUE THREAD

**Issue by radiatoryang** [ratwithacompiler/OBS-captions-plugin#10]
Great plugin, thanks so much for your work!... but I'm going to be that guy who makes a feature request.

Sometimes our stream uses a live human captioner who inputs text via StreamText.net, one of the most popular live human captioning solutions -- here's a demo example of that output: https://www.streamtext.net/player?event=IHaveADream

I've deconstructed the HTML viewer into a simple Node.JS thing... it polls a URL every X seconds, accumulates the JSON output, and then displays the last X lines. It works great in theory. Unfortunately I can't get OBS to feed the captions properly into Twitch now. https://github.com/EddieCameron/obs-websocket-streamtext-captions 

I have a feeling it'd be much better as a native OBS plugin... but I don't know C++.

Request: would you be able to add support for StreamText.net? In theory, I would assume this is similar to polling the Google API, except this is a public URL with no auth or API key needed. This would help out a lot of live events that do live streaming with human captioners but no easy way to pipe that through OBS, we're all banging our heads against this and resorting to open captions which aren't ideal.

(Thanks for your time!)

**Comment by ratwithacompiler** [comment]
I don't think adding other caption sources like that to this plugin really makes sense, a similar issue was brought up once already here with a bit mode detail: https://github.com/ratwithacompiler/OBS-captions-plugin/issues/4

Basically the short version is that it's much much easier to do that via OBS's scripting support (Lua or Python) or via `obs-websockets`. The only reason I had to go with a native C++ plugin was to be able to get audio from OBS directly which isn't possible with the other methods but the vast majority of my plugin code here is pretty tightly coupled to Google Speech API and mostly UI and API settings specific (I don't even use an actual HTTP lib) so adding other sources wouldn't be able to reuse much despite the similarity. 


In general if you just want to send captions out via OBS a native plugin is (in my opinion) total overkill and doesn't really give many advandages while being quite a bit more work and also harder to install for most users (scripts meanwhile can just be added and controlled via the OBS settings directly and let you have a basic and easy UI still). 


So I don't think I'll add that as it just doesn't seem suited to my plugin since it would basically be completely separate both code and UI wise and just seems better as a separate standalone plugin/tool/script to me (also UI confusion/usability wise).

Regarding your tool and the `obs-websocket` plugin, not sure if that was your issue but do note that the `SendCaptions` function of that plugin until very recently used `obs_output_output_caption_text1` in OBS which has a hardcoded minimum caption display duration of 2 seconds and if you send captions more often than that they just get queued up and fall behind farther and farther which made that not too useful.

That problem was fixed about two weeks ago (https://github.com/Palakis/obs-websocket/commit/0f6cbb0c5c300ec28b09e4408231b735e9f54f6d). There currently isn't a new release of it yet that includes those changes but with them it's possible to send captions at the same rate as a native plugin does and should be functionally equivalent now. Not sure how often they do releases but seems pretty regularly so that should probably be available soon and should hopefully fix any limitations you're facing with your tool.
